src/librustdoc/html/static/js/storage.js

```diff
@@ -24,7 +24,7 @@ function getSettingValue(settingName) {
             return def;
         }
     }
-    return current;
+    return current.replace(/[^A-Za-z0-9_-]/g,"");
```

Comment:
Can you add a comment explaining why this is needed and how this is fixed please for posterity?

Reply:
Done :+1:

Reply:
So looks good to me as is. Just one worry: if someone uses a non-latin letter (or any unicode character). For example a greek person adds a theme with a greek name for their personal use. It's not going to use then, right?

Reply:
That's true. Perhaps it's better to exclude a set of illegal characters instead? Something like `/[\s\(\)\\\/\*\"\'\`<>:;=&]/` would have the same effect, but I'd just want to be sure we're not forgetting a character we definitely don't want to get through.

Reply:
Apart from `<>.;=:()[]{}&|`, is there still a way to inject anything?

Reply:
Probably not! I feel comfortable enough going with that.

Reply:
True, although the whitespace stripping in the regex prevents the `new Array` case.

I didn't track down exactly where the injected code is being executed, but I suspect that somewhere we're pulling a theme setting out of localStorage and rendering it directly to the page (e.g. to indicate the currently selected theme/setting). Once the injected code is in the page content (e.g. `<svg onload=some_js_func()>`), refreshing the page causes it to be parsed as markup and executed.

It would be ideal to validate the values of each of the settings we store in localStorage, but that's a more involved change, and it's a good start to treat data pulled from localStorage as untrusted.

Reply:
I'd strongly recommend fixing the execution side of the problem instead of attempting to sanitize it. E.g. replacing `innerHTML` with `textContent` or whatever it's causing it.

Reply:
Ew, `document.write`. There's the `blocking="render"` attribute that can be set on stylesheets to solve just that. But it's fairly new and not yet supported by firefox. Maybe doing a sync XHR in the script to fetch the stylesheet (so it's guaranteed to be in cache) could be used to  block rendering.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/link#blocking

Reply:
A `FIXME` then?

Reply:
I just pushed a quick commit that attempts to ensure the theme is among the actual defined themes, using the same logic used in `buildSettingsPage`. It also URI encodes the theme name when inserting into the CSS `<link>`.